### PR TITLE
make pd start timeout configurable

### PR DESCRIPTION
**Pull request by oliviachenairbnb**
### What problem does this PR solve?

make pd start timeout configurable
### What is changed and how does it work?

Update pd start timeout value to make it configurable 
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/manager/member/startscript/v2/pd_start_script.go [hunk 1]
```diff
@@ -77,13 +80,14 @@ const (
...
 PD_DOMAIN={{ .PDDomain }}
+threshold={{ .PDStartTimeout }} 
 
 elapseTime=0
 period=1
-threshold=30
 while true; do
     sleep ${period}
     elapseTime=$(( elapseTime+period ))
+	echo "waiting threshold for pd ${threshold}"
```
Comment (csuzhangxc) [inline pkg/manager/member/startscript/v2/pd_start_script.go:90]: update the content of this script (in a ConfigMap) will cause existing PD rolling to restart. How about just replacing `threshold=30` with `threshold={{ .PDStartTimeout }}`?

Comment (oliviachenairbnb) [reply pkg/manager/member/startscript/v2/pd_start_script.go]: updated. can you pls help review again? thx

**Comment by oliviachenairbnb** [comment]
saw ci failing due to unrelated problems. do you have any insights how to pass the ci check? thanks. @csuzhangxc

**Comment by csuzhangxc** [comment]
> saw ci failing due to unrelated problems. do you have any insights how to pass the ci check? thanks. @csuzhangxc

"pd_start_script_test.go:558: unexpected (-want, +got):   strings.Join({" I think it's related. @oliviachenairbnb

**Comment by oliviachenairbnb** [comment]
fixed the failing ci. can you pls help review again? thx @csuzhangxc

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**